### ISSUE TITLE
6.0.2 AsyncValidation fails with `Uncaught (in promise)` on failed validation

### ISSUE THREAD

**Issue by vitalis** [erikras/redux-form#1704]
here is a code from my validator: (works in 5.3.3)
```javascript
export function imageSize(fieldName, config) {
  if (!config) throw new Error('Must provide config to imageSize.');

  return (values) => {
    const file = values[fieldName];
    if (!file) return Promise.resolve();

    return loadImage(file)
      .then((image) => {
        if (!isImageInSize(image, config)) {
          throw `image must be ${config.width}x${config.height}.`;
        }
      })
      .catch((error) => {
        throw { [fieldName]: error };
      });
  };
}
```

In 6.0.2 I have the following error on failed validation:
<img width="474" alt="apprime" src="https://cloud.githubusercontent.com/assets/36814/18253079/a80b923c-739a-11e6-83f6-e5596034da37.png">

**Comment by rafusmx** [comment]
I have the same issue. Everything worked fine until we started working on the data validations. It gives me the impression that there is something missing on the throw calls, but we have not been able to figure it out.

**Comment by erikras** [comment]
Fix published in [`v6.0.3`](https://github.com/erikras/redux-form/releases/tag/v6.0.3).
